firefox_ui_tests/remote/security/test_safe_browsing_warning_pages.py

```diff
@@ -0,0 +1,116 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette import By
+from marionette.errors import NoSuchElementException
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestSafeBrowsingWarningPages(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.urls = [
+            # Phishing URL
+            "https://www.itisatrap.org/firefox/its-a-trap.html",
+            # Malware URL
+            "https://www.itisatrap.org/firefox/its-an-attack.html"
+            ]
+
+        self.prefs.set_pref('browser.safebrowsing.enabled', True)
+        self.prefs.set_pref('browser.safebrowsing.malware.enabled', True)
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        time.sleep(2)
```

Comment:
Oh! So the init method itself gets called with a 2s timeout. Adding the moz entries will then happen in 2s+Xms here:
https://dxr.mozilla.org/mozilla-central/source/toolkit/components/url-classifier/SafeBrowsing.jsm?from=SafeBrowsing.jsm#199

Given that there is no observer we could listen for we should at least use 3s as sleep here. With 2s only we could run into race conditions.